errorVarsName/errorVars.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package errorVarsName
+
+import (
+	"go/ast"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+const (
+	appErrorString = "*github.com/mattermost/mattermost-server/v5/model.AppError"
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name: "errorVarsName",
+	Doc:  "check for err and appErr var names corresponding to the expected types",
+	Run:  run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	for _, file := range pass.Files {
+		ast.Inspect(file, func(node ast.Node) bool {
+			switch x := node.(type) {
+			case *ast.AssignStmt:
+				for idx, lhsExpr := range x.Lhs {
+					idnt, ok := lhsExpr.(*ast.Ident)
+					if !ok {
+						return true
+					}
+
+					if strings.HasPrefix(idnt.Name, "err") {
+						if len(x.Rhs) == 1 {
+							if typeAndValue, ok := pass.TypesInfo.Types[x.Rhs[0]]; ok {
+								returnTypes := strings.Split(strings.Trim(typeAndValue.Type.String(), "()"), ", ")
```

Comment:
Same comment request that was done in #16

Reply:
Done